folium/plugins/minimap.py

```diff
@@ -0,0 +1,130 @@
+    Uses the Leaflet plugin by Norkart under BSD 2-Clause "Simplified" License.
+    https://github.com/Norkart/Leaflet-MiniMap
+
+    Parameters
+    ----------
+    tile_layer: folium TileLayer object or str, optional
+        Provide a folium TileLayer object or the wanted tiles as string.
+        If not provided it will use the default of `TileLayer`.
+
+    position : str, default 'bottomright'
+        The standard Control position parameter for the widget.
+
+    width: int, default 150
+        The width of the minimap in pixels.
+
+    height: int, default 150
+        The height of the minimap in pixels.
+
+    collapsed_width: int, default 25
+        The width of the toggle marker and the minimap when collapsed, in pixels.
+
+    collapsed_height: int, default 25
+        The height of the toggle marker and the minimap when collapsed
+
+    zoom_level_offset: int, defalaut -5
+        The offset applied to the zoom in the minimap compared to the zoom of the main map. Can be positive or negative.
+
+    zoom_level_fixed: int, default None
+        Overrides the offset to apply a fixed zoom level to the minimap regardless of the main map zoom.
+        Set it to any valid zoom level, if unset zoomLevelOffset is used instead.
+
+    center_fixed: bool, default False
+        Applies a fixed position to the minimap regardless of the main map's view / position.
+        Prevents panning the minimap, but does allow zooming (both in the minimap and the main map).
+        If the minimap is zoomed, it will always zoom around the centerFixed point.
+        You can pass in a LatLng-equivalent object.
+
+    zoom_animation: bool, default False
+        Sets whether the minimap should have an animated zoom.
+        (Will cause it to lag a bit after the movement of the main map.)
+
+    toggle_display: bool, default False
+        Sets whether the minimap should have a button to minimise it.
+
+    auto_toggle_display: bool, default False
+        Sets whether the minimap should hide automatically
+        if the parent map bounds does not fit within the minimap bounds.
+        Especially useful when 'zoomLevelFixed' is set.
+
+    minimized: bool, default False
+        Sets whether the minimap should start in a minimized position.
+
+    Examples
+    --------
+    >>> MiniMap(tile_layer='Stamen WaterColor', position='bottomleft')
+    """
+
+    _template = Template("""
+        {% macro script(this, kwargs) %}
+
+        var {{ this.tile_layer.get_name() }} = L.tileLayer(
+        '{{ this.tile_layer.tiles }}',
+        {{ this.tile_layer.options }} );
+
+        var {{ this.get_name() }} = new L.Control.MiniMap( {{this.tile_layer.get_name()}},
+         {{ this.options }});
+        {{ this._parent.get_name() }}.addControl({{ this.get_name() }});
+
+        {% endmacro %}
+    """)  # noqa
+
+    def __init__(self, tile_layer=None, position='bottomright', width=150,
+                 height=150, collapsed_width=25, collapsed_height=25,
+
```

Comment:
Empty line

Reply:
fixed

Reply:
Regarding the optional requirement above,  It defaults to to whatever TileLayer defaults to (OSM I think) but I'm not sure I should hard code this in.  So I put this in which might have been missing when you looked:
https://github.com/talbertc-usgs/folium/blob/minimap/folium/plugins/minimap.py#L23